src/psyclone/psyGen.py

```diff
@@ -6093,50 +6198,71 @@ def __init__(self, kernel=None):
         # Reference to KernelSchedule to which this symbol table belongs.
         self._kernel = kernel
 
-    def declare(self, name, datatype, shape=[], scope='local',
-                is_input=False, is_output=False):
-        '''
-        Declare a new symbol in the symbol table.
-
-        :param str name: Name of the symbol.
-        :param str datatype: Datatype of the symbol.
-        :param list shape: Shape of the symbol in column-major order \
-                           (leftmost index is contiguous in memory). Each \
-                           entry represents an array dimension. If it is \
-                           'None' the extent of that dimension is unknown, \
-                           otherwise it holds an integer literal or a \
-                           reference to an integer symbol with the extent. \
-                           If it is an empty list then the symbol represents \
-                           a scalar.
-        :param str scope: It is 'local' if the symbol just exists inside the \
-                          kernel scope or 'global_*' if the data survives \
-                          outside of the kernel scope. Note that \
-                          global-scoped symbols also have postfixed \
-                          information about the sharing mechanism, at the \
-                          moment just 'global_argument' is available for \
-                          variables passed in/out of the kernel by argument.
-        :param bool is_input: Whether the symbol represents data that exists \
-                              before the kernel is entered and that is passed \
-                              into the kernel.
-        :param bool is_output: Whether the symbol represents data that is \
-                               survives outside the kernel upon exit.
-        :raises KeyError: The provided name can not be used as key in the
-                          table.
-        '''
-        if name in self._symbols:
-            raise KeyError("Symbol table already contains a symbol with"
-                           " name '{0}'.".format(name))
+    def declare(self, new_symbol):
```

Comment:
Suggestion (and a very picky one): Would now `symbol_table.insert(symbol)` fit better than `symbol_table.declare(symbol)`? This would require string substitution in several files so I am happy with keeping 'declare', what are your thoughts?

Reply:
After discussion we agreed to go with `add`.